src/riak_kv_ttaaefs_manager.erl

```diff
@@ -725,13 +962,45 @@ generate_repairfun(ExchangeID, QueueName) ->
                 end
             end,
         {ToRepair, SinkDCount} = lists:foldl(FoldFun, {[], 0}, RepairList),
-        lager:info("AAE exchange ~w shows sink ahead for ~w keys", 
-                    [ExchangeID, SinkDCount]),
-        lager:info("AAE exchange ~w outputs ~w keys to be repaired",
-                    [ExchangeID, length(ToRepair)]),
+        lager:info("AAE reqid=~w work_item=~w type=~w shows sink ahead " ++
+                        "for key_count=~w keys limited by max_results=~w", 
+                    [ExchangeID, WorkType, Ref, SinkDCount, MaxResults]),
         riak_kv_replrtq_src:replrtq_ttaaefs(QueueName, ToRepair),
-        lager:info("AAE exchange ~w has requeue complete for ~w keys",
-                    [ExchangeID, length(ToRepair)])
+        PBDL = report_repairs(ExchangeID, ToRepair, Ref, WorkType),
+        RB = app_helper:get_env(riak_kv, ttaaefs_rangeboost, ?RANGE_BOOST),
+        TargetForRange = 
+            case WorkType of
+                range_check ->
+                    MaxResults div RB;
+                _AltCheck ->
+                    MaxResults div 2
+            end,
+        case length(ToRepair) > TargetForRange of
+            true ->
+                % If there is a single bucket, use that as a constraint,
+                % otherwise query over all buckets on the given modified
+                % range
+                case PBDL of
+                    [{B, KC, LowDT, HighDT}] ->
+                        lager:info("Setting range to bucket=~p ~w ~w " ++
+                                    " last_count=~w set by work_item=~w",
+                                    [B, LowDT, HighDT, KC, WorkType]),
+                        set_range(B, all, LowDT, HighDT);
+                    _ ->
+                        MapFun = fun({_B, _KC, LD, HD}) -> {LD, HD} end,
```

Comment:
if you only want a LoDT and HiDT, then this code can be replaced by a fold.
```
{LoDT, HiDT} = 
      lists:foldl(fun({_B, _KC, LD, HD}, {Min, Max}) ->
                               {case LD < Min of true -> LD; false Min end,
                               {case HD > Max of true -> HD; false Max end}
                 end, {0,0}, PBDL),

Reply:
I this way yo don't need the case either, because if you have one element it's just a special case

Reply:
I'm OK with how it reads at the moment.  I'm minded to leave as-is for now.